packages/block-editor/src/components/block-list/block-wrapper.js

```diff
@@ -28,148 +28,160 @@ import { Context, SetBlockNodes } from './root-container';
-			'core/block-editor'
-		);
-		const fallbackRef = useRef();
-		wrapper = wrapper || fallbackRef;
-
-		const [ isHovered, setHovered ] = useState( false );
-
-		// Provide the selected node, or the first and last nodes of a multi-
-		// selection, so it can be used to position the contextual block toolbar.
-		// We only provide what is necessary, and remove the nodes again when they
-		// are no longer selected.
-		useEffect( () => {
-			if ( isSelected || isFirstMultiSelected || isLastMultiSelected ) {
-				const node = wrapper.current;
-				setBlockNodes( ( nodes ) => ( {
-					...nodes,
-					[ clientId ]: node,
-				} ) );
-				return () => {
-					setBlockNodes( ( nodes ) => omit( nodes, clientId ) );
-				};
-			}
-		}, [ isSelected, isFirstMultiSelected, isLastMultiSelected ] );
-
-		// translators: %s: Type of block (i.e. Text, Image etc)
-		const blockLabel = sprintf( __( 'Block: %s' ), blockTitle );
-
-		// Handing the focus of the block on creation and update
-
-		/**
-		 * When a block becomes selected, transition focus to an inner tabbable.
-		 */
-		const focusTabbable = () => {
-			// Focus is captured by the wrapper node, so while focus transition
-			// should only consider tabbables within editable display, since it
-			// may be the wrapper itself or a side control which triggered the
-			// focus event, don't unnecessary transition to an inner tabbable.
-			if (
-				document.activeElement &&
-				isInsideRootBlock( wrapper.current, document.activeElement )
-			) {
-				return;
-			}
-
-			// Find all tabbables within node.
-			const textInputs = focus.tabbable
-				.find( wrapper.current )
-				.filter( isTextField )
-				// Exclude inner blocks and block appenders
-				.filter(
-					( node ) =>
-						isInsideRootBlock( wrapper.current, node ) &&
-						! node.closest( '.block-list-appender' )
-				);
-
-			// If reversed (e.g. merge via backspace), use the last in the set of
-			// tabbables.
-			const isReverse = -1 === initialPosition;
-			const target =
-				( isReverse ? last : first )( textInputs ) || wrapper.current;
-
-			placeCaretAtHorizontalEdge( target, isReverse );
-		};
-
-		useEffect( () => {
-			if ( shouldFocusFirstElement ) {
-				focusTabbable();
-			}
-		}, [ shouldFocusFirstElement ] );
+/**
+ * This hook is used to lighly mark an element as a block element. Call this
+ * hook and pass the returned props to the element to mark as a block. If you
+ * define a ref for the element, it is important to pass the ref to this hook,
+ * which the hooks in turn will pass to the component through the props it
+ * returns. Optionally, you can also pass any other props through this hook, and
+ * they will be merged and returned.
+ *
+ * @param {Object}  props   Optional. Props to pass to the element. Must contain
+ *                          the ref if one is defined.
+ * @param {Object}  options Options for internal use only.
+ * @param {boolean} options.__unstableIsHtml
```

Comment:
Why put the unstable props in a separate parameter? They're already labeled with the `__unstable` prefix.

Reply:
The linter logs a warning if it's not documented.

Reply:
No, I mean why is it under a second `options` object, rather than the same one that contains the other parameters? Why is the function signature `useBlockWrapperProps( { ref }, { __unstableIsHtml } )` rather than `useBlockWrapperProps( { ref, __unstableIsHtml } )`?

Reply:
I believe because the first one is "props" to be merged with the output and the second one is settings for the hook itself.

Reply:
Oh, okay. I get it now. :+1: